### PR TITLE
concourse: provide a minimal api to interact with the e2e tests

### PR DESCRIPTION
**Pull request by MaximilianMeister**
* trigger a build
* fetch the result
* determine if test is running

## Depends on https://github.com/kubic-project/e2e-tests/pull/3

NOTE: this will probably be run on one of our workers and serve as an interface for the ci job

### CODE REVIEW

**Comment by jimmykarily** [comment]
@MaximilianMeister #3 is merged. You can rebase this one.

File: api/api.rb [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+      Pathname.new(settings.root).join("e2e-result.json")
+    end
+
+    def json_result
+      JSON.parse(result_file_path.read)
+    end
+  end
+
+  post '/start' do
+    if result_file_path.size.zero?
```
Comment (jimmykarily) [inline api/api.rb]:
```
- -> /start
2017-02-28 13:42:58 - Errno::ENOENT - No such file or directory @ rb_file_s_size - /home/dimitris/workspace/suse/e2e-tests/e2e-result.json:
	api/api.rb:21:in `size'
	api/api.rb:21:in `size'
	api/api.rb:21:in `block in <class:RspecResultApi>'
```
when I do:

```
$ curl -X POST http://localhost:4567/start -d ''
```

Comment (MaximilianMeister) [inline api/api.rb]: oh right, this happens when it hasn't ran at least once... fixed

**Comment by jimmykarily** [comment]
👍

File: api/api.rb [hunk 2]
```diff
@@ -0,0 +1,50 @@
+require "sinatra/base"
+require "sinatra/json"
+
+require "json"
+require "pathname"
+
+class RspecResultApi < Sinatra::Base
+  set :root, File.expand_path("../..", __FILE__)
+  set :bind, "0.0.0.0"
```
Comment (MaximilianMeister) [inline api/api.rb]: had to add one more, this small change

**Pull request merged**